### PR TITLE
[linky] One unique method to update daily and weekly data channels

### PR DESCRIPTION
**Pull request by lolodomo**
Also check data availability for the peak consumption of yesterday
Also log power peak data

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@clinique : first, this PR avoids calling a request twice when data from yesterday are not yet available in the morning. Second, it applies the data check availability for the yesterday peak consumption (I forgot this channel in my previopus improvement).

**Comment by lolodomo** [comment]
@clinique : can you please have a look, the change is small ;)

**Comment by clinique** [comment]
@lolodomo : I did, please see my 2 comments.

**Comment by lolodomo** [comment]
@clinique : I see no comment from you ! Are you sure you submitted your review comments ?

File: bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java [hunk 1]
```diff
@@ -502,19 +493,29 @@ public void checkData(Consumption consumption) throws LinkyException {
...
     private boolean isDataLastDayAvailable(Consumption consumption) {
         Aggregate days = consumption.aggregats.days;
-        logData(days, "Last day", false, DateTimeFormatter.ISO_LOCAL_DATE, true);
+        logData(days, "Last day", false, DateTimeFormatter.ISO_LOCAL_DATE, false, true);
         return days.datas != null && days.datas.size() > 0 && !days.datas.get(days.datas.size() - 1).isNaN();
     }
 
     private void logData(Aggregate aggregate, String title, boolean withDateFin, DateTimeFormatter dateTimeFormatter,
-            boolean onlyLast) {
+            boolean onlyFirst, boolean onlyLast) {
```
[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java:509]: Yes, you"re right. Enum should be FIRST, LAST or ALL.

[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java]: Done

File: bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java [hunk 2]
```diff
@@ -461,14 +447,19 @@ public synchronized void handleCommand(ChannelUID channelUID, Command command) {
         }
     }
 
-    private @Nullable Consumption getConsumptionAfterChecks(Consumption consumption) {
+    private @Nullable Consumption getConsumptionAfterChecks(Consumption consumption, boolean checkFirst,
```
[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java]: Done

**Review by clinique — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
@cpmeister : would appreciate if you could review this one.

**Review by Hilbrand — APPROVED** [review]
LGTM

**Pull request merged**